packages/admin-ui/src/page/page-header.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react'
+import type { ComponentPropsWithRef } from 'react'
+import { createComponent, useElement } from '@vtex/admin-ui-react'
+
+import { PageHeaderContext } from './page-header-context'
+import * as style from './page.style'
+
+/**
+ * Page header component
+ *
+ * @example
+ * import { Tabs, TabPanel, Button, PageHeader } from "@vtex/admin-ui"
+ *
+ * <Tabs state={tabs}>
+ *   <PageHeader onPopNavigation={() => alert('onPopNavigation()')}>
+ *     Product #123
+ *   </PageHeader>
+ *   <TabPanel id="1">
+ *     <Button onClick={() => tabs.select('3')}>Go to Tab 3!</Button>
+ *   </TabPanel>
+ *   <TabPanel id="2">
+ *     <Button onClick={() => tabs.select('1')}>Go to Tab 1!</Button>
+ *   </TabPanel>
+ *   <TabPanel id="3">
+ *     <Button onClick={() => tabs.select('2')}>Go to Tab 2!</Button>
+ *   </TabPanel>
+ * </Tabs>
+ */
+export const PageHeader = createComponent<'header', PageHeaderOptions>(
+  (props) => {
+    const { onPopNavigation, children, ...htmlProps } = props
+
+    const [scrollOnTop, setScrollOnTop] = useState(window.scrollY === 0)
+
+    useEffect(() => {
+      const threshold = 0
+      let lastScrollY = window.scrollY
+      let ticking = false
+
+      const updateScrolling = () => {
+        const scrollY = window.scrollY
+
+        if (scrollY === threshold) {
+          setScrollOnTop(true)
+
+          return
+        }
+
+        if (Math.abs(scrollY - lastScrollY) < threshold) {
+          ticking = false
+
+          return
+        }
+
+        setScrollOnTop(false)
+        lastScrollY = scrollY > threshold ? scrollY : threshold
+        ticking = false
+      }
+
+      const onScroll = () => {
```

Comment:
Was this created to add a box-shadow when the user scrolls down the page? 🤔 If the answer is yes, i think it could be better to use the [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API).

Reply:
Good catch! Yes, that's the expected behavior!

Reply:
I decided not to go with the Intersection Observer API because it is mainly used for complex scrolling behaviors, such as infinite scrolls and elements that appear when a certain amount of the page has been scrolled. Since this is such a simple behavior, it occurs to me that an event listener could be the best approach. Do you see any downsides with the event listener approach vs intersection observer? cc @Lenderson

Reply:
Thanks for the tip, Lenderson! I developed a mixed approach using the Intersection Observer API + Portals. Indeed, it's unusual to see this use case being handled by the IO API, but nonetheless it is the best approach, as you pointed 😉